File: 3105-longest-strictly-increasing-or-strictly-decreasing-subarray/3105-longest-strictly-increasing-or-strictly-decreasing-subarray.py
class Solution:
  def longestMonotonicSubarray(self, nums: List[int]) -> int:
    inc, dec, ans = 1, 1, 1
    for i in range(1, len(nums)):
      inc = inc + 1 if nums[i] > nums[i-1] else 1
      dec = dec + 1 if nums[i] < nums[i-1] else 1
      ans = max(ans, inc, dec)
    return ans
        

File: 3105-longest-strictly-increasing-or-strictly-decreasing-subarray/3105-longest-strictly-increasing-or-strictly-decreasing-subarray.go
func longestMonotonicSubarray(nums []int) int {
  inc, dec, ans := 1, 1, 1
  for i := 1; i < len(nums); i++ {
    if nums[i] > nums[i-1] {
      inc++
      dec = 1
      ans = max(ans, inc)
    } else if nums[i] < nums[i-1] {
      dec++
      inc = 1
      ans = max(ans, dec)
    } else if inc > 1 || dec > 1 {
      inc = 1
      dec = 1
    }
  }
  return ans
}